### PR TITLE
Bug 1224009 - migrate system app to NotificationHelper

### PR DESCRIPTION
**Pull request by hkasemir**
https://bugzilla.mozilla.org/show_bug.cgi?id=1224009

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26581894c350d7aba3350b1712610ce76916ece5

File: apps/system/js/captive_portal.js [hunk 1]
```diff
@@ -74,12 +76,13 @@ var CaptivePortal = {
...
+    NotificationHelper.send('', options).then(notification => {
+      notification.addEventListener('click',
+        this.captiveNotification_onClick(notification));
+      notification.addEventListener('close', () => {
+        notification = null;
+      });
+    });
   },
 
   dismissNotification: function dismissNotification(id) {
```
Comment (hkasemir) [inline apps/system/js/captive_portal.js]: There are a bunch of calls to `this.notification` and I'm not sure how to resolve those in the `dismissNotification` function. The old code assigned the new Notification to `this.notification`, but that doesn't seem to work with the `NotificationHelper`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a2bfacb6c583e314533a65fe62887fcfd56713c1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a95098461a61dffe69b7fb43a322629557447fd0

File: apps/system/js/captive_portal.js [hunk 2]
```diff
@@ -20,7 +21,6 @@ var CaptivePortal = {
 
   handleLogin: function cp_handleLogin(id, url) {
     var wifiManager = window.navigator.mozWifiManager;
-    var _ = window.navigator.mozL10n.get;
     var settings = window.navigator.mozSettings;
     var icon = window.location.protocol + '//' + window.location.hostname +
```
Comment (sfoster) [inline apps/system/js/captive_portal.js]: drive by suggestion: window.location.origin

File: apps/system/js/captive_portal.js [hunk 3]
```diff
@@ -51,22 +53,22 @@ var CaptivePortal = {
       return;
     }
 
-    this.captiveNotification_onClick = (function() {
-      this.notification.removeEventListener('click',
-                                            this.captiveNotification_onClick);
+    this.captiveNotification_onClick = notification => {
+      notification.removeEventListener('click',
```
Comment (sfoster) [inline apps/system/js/captive_portal.js]: I think this should be notification.removeEventListener(this.captiveNotification_onClick) - so you remove the function reference, not the result of a call to it

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=208e589b3754961a4786bf90d7bbe9bda38dbd66

File: build/l10n/xfail.list [hunk 4]
```diff
@@ -96,7 +91,6 @@ apps/system/js/device_storage_watcher.js
 apps/system/js/update_manager.js
 apps/system/js/value_selector/spin_date_picker.js
 apps/system/js/value_selector/value_selector.js
-apps/system/js/voicemail.js
```
Comment (zbraniecki) [inline build/l10n/xfail.list]: you're not altering this file in your patch, you shouldn't remove it from here.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9fd7d4c45e572c23b3c26608b377766ffe98bd3d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9dd513d1adfd4bc3a55f41dfeb61849ab58b728

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9dd513d1adfd4bc3a55f41dfeb61849ab58b728

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b8c0a25868d65426346b9f6f3021dc7ea23f4b85

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=22656faa2cac32d8f20a3b5ebfcf533450d4a5a0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c22c9e7f6c790f55831d4356d985380f81f68c8f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5411f24e5a8ddd96e7f822e4ce04bbc9f44beab7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=045aaea83a2afdb7b5650a9fc0d3c0713802fcf7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6ad26f4c624832d87a0f9840608033b2cacd8eef

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0a013e7ba66b9fe2d467ea53849b2d7ddae622b1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0f804a6af9a6ce3b9154061491d690484fe851aa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d7c7628f902aa4e47b7e139b48dab2f06772c9fd

**Pull request merged**